apps/rush-lib/src/logic/PackageChangeAnalyzer.ts

```diff
@@ -92,7 +110,16 @@ export class PackageChangeAnalyzer {
         | RushConfigurationProject
         | undefined = this._rushConfiguration.findProjectForPosixRelativePath(filePath);
       if (owningProject) {
-        projectHashDeps.get(owningProject.packageName)!.set(filePath, fileHash);
+        const relativePath: string = path.relative(owningProject.projectRelativeFolder, filePath);
+        const ignoreMatchers: IMinimatch[] | undefined = ignorePatternsByProject.get(
```

Comment:
We already have a dependency on `ignore`, which can run this like it is processing a .gitignore file and theoretically optimize a bit.